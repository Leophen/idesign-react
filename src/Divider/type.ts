export interface DividerProps {
  /**
   * 文本位置
   * @default center
   */
  align?: 'left' | 'center' | 'right';
  /**
   * 类名
   */
  className?: string;
  /**
   * 内容
   */
  children?: React.ReactNode;
  /**
   * 是否为虚线
   * @default false
   */
  dashed?: boolean;
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
}
