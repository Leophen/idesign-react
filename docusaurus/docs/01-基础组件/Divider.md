---
title: Divider 分割线
---

分割线是一个呈线状的轻量化组件，起到分割、组织、细化的作用，用于有逻辑的组织元素内容和页面结构。

## 基本用法

可通过 `dashed` 属性来控制分割线样式。

```tsx live
<div>
  <p>
  柯里化即Currying，是一门编译原理层面的技术，用途是实现多参函数，其为实现多参函数提供了一个递归降解的实现思路——把接受多个参数的函数变换成接受第一个参数的函数，并且返回接受剩余参数且返回结果的新函数。
  </p>
  <Divider></Divider>
  <p>
  柯里化即Currying，是一门编译原理层面的技术，用途是实现多参函数，其为实现多参函数提供了一个递归降解的实现思路——把接受多个参数的函数变换成接受第一个参数的函数，并且返回接受剩余参数且返回结果的新函数。
  </p>
  <Divider dashed></Divider>
  <p>
  柯里化即Currying，是一门编译原理层面的技术，用途是实现多参函数，其为实现多参函数提供了一个递归降解的实现思路——把接受多个参数的函数变换成接受第一个参数的函数，并且返回接受剩余参数且返回结果的新函数。
  </p>
</div>
```

## 带文字的分割线

可通过 `align` 属性来控制分割线文字内容的位置。

```tsx live
<div>
  <p>
    柯里化即Currying，是一门编译原理层面的技术，用途是实现多参函数，其为实现多参函数提供了一个递归降解的实现思路——把接受多个参数的函数变换成接受第一个参数的函数，并且返回接受剩余参数且返回结果的新函数。
  </p>
  <Divider align="left">iDesign</Divider>
  <p>
    柯里化即Currying，是一门编译原理层面的技术，用途是实现多参函数，其为实现多参函数提供了一个递归降解的实现思路——把接受多个参数的函数变换成接受第一个参数的函数，并且返回接受剩余参数且返回结果的新函数。
  </p>
  <Divider align="center">iDesign</Divider>
  <p>
    柯里化即Currying，是一门编译原理层面的技术，用途是实现多参函数，其为实现多参函数提供了一个递归降解的实现思路——把接受多个参数的函数变换成接受第一个参数的函数，并且返回接受剩余参数且返回结果的新函数。
  </p>
  <Divider align="right">iDesign</Divider>
  <p>
    柯里化即Currying，是一门编译原理层面的技术，用途是实现多参函数，其为实现多参函数提供了一个递归降解的实现思路——把接受多个参数的函数变换成接受第一个参数的函数，并且返回接受剩余参数且返回结果的新函数。
  </p>
</div>
```

## API

| 属性      | 说明       | 类型                        | 默认值     |
| --------- | ---------- | --------------------------- | ---------- |
| align     | 文本位置   | `"left"\|"center"\|"right"` | `"center"` |
| className | 类名       | `string`                    | `--`       |
| children  | 内容       | `ReactNode`                 | `--`       |
| dashed    | 是否为虚线 | `boolean`                   | `false`    |
| style     | 自定义样式 | `CSSProperties`             | `--`       |
