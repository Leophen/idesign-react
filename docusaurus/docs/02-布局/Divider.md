---
title: Divider 分割线
---

区隔内容的分割线。

## 引入

```js
import {Divider} from 'idesign';
```

## 基本用法

默认为水平分割线，可在中间加入文字。

```jsx live fffx
<>
    <p>
        对不同章节的文本段落进行分割。对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
    <Divider />
    <p>
        对不同章节的文本段落进行分割。对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
    <Divider dashed />
    <p>
        对不同章节的文本段落进行分割。对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
</>
```

## 垂直分割线

使用 `type="vertical"` 设置为行内的垂直分割线。

```jsx live fffx
<>
    Text
    <Divider type="vertical" />
    <a href="#">Link</a>
    <Divider type="vertical" />
    <a href="#">Link</a>
</>
```

## 带文字的分割线

分割线中带有文字，可以用 `orientation` 指定文字位置。

```jsx live fff
<>
    <p>
        对不同章节的文本段落进行分割。对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
    <Divider>Text</Divider>
    <p>
        对不同章节的文本段落进行分割。对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
    <Divider orientation="left">Left Text</Divider>
    <p>
        对不同章节的文本段落进行分割。对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
    <Divider orientation="right">Right Text</Divider>
    <p>
        对不同章节的文本段落进行分割。对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
    <Divider orientation="left" orientationMargin="0">
        Left Text with 0 orientationMargin
    </Divider>
    <p>
        对不同章节的文本段落进行分割。对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
    <Divider orientation="right" orientationMargin={50}>
        Right Text with 50px orientationMargin
    </Divider>
    <p>
        对不同章节的文本段落进行分割。对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
</>
```

## 分割文字使用正文样式

使用 `plain` 可以设置为更轻量的分割文字样式。

```jsx live fff
<>
    <p>
        对不同章节的文本段落进行分割。 对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
    <Divider plain>Text</Divider>
    <p>
        对不同章节的文本段落进行分割。 对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
    <Divider orientation="left" plain>
        Left Text
    </Divider>
    <p>
        对不同章节的文本段落进行分割。 对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
    <Divider orientation="right" plain>
        Right Text
    </Divider>
    <p>
        对不同章节的文本段落进行分割。 对行内文字/链接进行分割，例如表格的操作列。对不同章节的文本段落进行分割。
        对行内文字/链接进行分割，例如表格的操作列。
    </p>
</>
```

## API

| 参数              | 说明                                                                                               | 类型                      | 默认值       |
| ----------------- | -------------------------------------------------------------------------------------------------- | ------------------------- | ------------ |
| children          | 嵌套的标题                                                                                         | `ReactNode`               | `--`         |
| className         | 分割线样式类                                                                                       | `string`                  | `--`         |
| dashed            | 是否虚线                                                                                           | `boolean`                 | `false`      |
| orientation       | 分割线标题的位置                                                                                   | `left`〡`right`〡`center` | `center`     |
| orientationMargin | 标题和最近 left/right 边框之间的距离，去除了分割线<br/>同时 `orientation` 必须为 `left` 或 `right` | `string`〡`number`        | `--`         |
| plain             | 文字是否显示为普通正文样式                                                                         | `boolean`                 | `false`      |
| style             | 分割线样式对象                                                                                     | `CSSProperties`           | `--`         |
| type              | 水平还是垂直类型                                                                                   | `horizontal`〡`vertical`  | `horizontal` |

