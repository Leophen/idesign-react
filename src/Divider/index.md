---
nav:
  title: 组件
  path: /components
group:
  title: 基础组件
  order: 1
order: 3
---

# Divider 分割线（开发中）

## Demo

```tsx
import React from 'react';
import { Divider } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-column">
      <div className="idesign-demo-block-row">
        <Divider>填充按钮</Divider>
      </div>
    </div>
  );
};

export default App;
```
