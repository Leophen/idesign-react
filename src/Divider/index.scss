.i-divider {
  box-sizing: border-box;
  margin: 24px 0;
  color: #303133;
  font-size: 14px;
  border-top: 1px solid #e5e7eb;
}

.i-divider--dashed {
  border-top-style: dashed;
}

.i-divider--with-text {
  display: flex;
  white-space: nowrap;
  border-top: 0;
  &::before,
  &::after {
    position: relative;
    top: 50%;
    width: 50%;
    border-top: 1px solid #e5e7eb;
    transform: translateY(calc(50% - 0.5px));
    content: '';
  }
}

.i-divider--with-text-left {
  &::before {
    width: 5%;
  }
  &::after {
    width: 95%;
  }
}

.i-divider--with-text-right {
  &::after {
    width: 5%;
  }
  &::before {
    width: 95%;
  }
}

.i-divider--text {
  display: inline-block;
  padding: 0 1em;
}
