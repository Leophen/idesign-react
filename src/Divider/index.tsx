import React from 'react';
import classNames from 'classnames';
import './index.scss';

export interface DividerProps {
  /**
   * 文本位置
   * @default center
   */
  align?: 'left' | 'right' | 'center';
  /**
   * 类名
   */
  className?: string;
  /**
   * 内容
   */
  children?: React.ReactNode;
  /**
   * 是否为虚线
   * @default false
   */
  dashed?: boolean;
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
}

const Divider: React.FC<DividerProps> = (props) => {
  const { align = 'center', children, className, dashed = false, style, ...restProps } = props;

  return (
    <div
      className={classNames(
        'i-divider',
        dashed && 'i-divider--dashed',
        children && 'i-divider--with-text',
        children && `i-divider--with-text-${align}`,
        className,
      )}
      style={{ ...style }}
      {...restProps}
    >
      {children && <span className="i-divider--text">{children}</span>}
    </div>
  );
};

Divider.displayName = 'Divider';

export default Divider;
