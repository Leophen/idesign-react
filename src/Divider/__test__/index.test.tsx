import { render } from '@testing-library/react';
import Divider from '../index';

describe('Divider 组件测试', () => {
  it('create', () => {
    const { asFragment } = render(<Divider />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('default', () => {
    const { container } = render(<Divider />);
    expect(container.firstChild).toHaveClass('i-divider');
  });

  it('dashed', () => {
    const { container } = render(<Divider dashed />);
    expect(container.firstChild).toHaveClass('i-divider--dashed');
  });

  it('align', () => {
    const { container, getByText } = render(<Divider align="left">iDesign</Divider>);
    expect(container.firstChild).toHaveClass('i-divider--with-text-left');
    expect(getByText('iDesign').textContent).toBe('iDesign')
  });
});

