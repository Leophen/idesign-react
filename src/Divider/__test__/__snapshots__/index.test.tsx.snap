// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider 组件测试 create 1`] = `
<DocumentFragment>
  <div
    class="i-divider"
  />
</DocumentFragment>
`;
